pydantic/json_schema.py

```diff
@@ -128,6 +128,34 @@ class PydanticJsonSchemaWarning(UserWarning):
 CoreModeRef = Tuple[CoreRef, JsonSchemaMode]
 JsonSchemaKeyT = TypeVar('JsonSchemaKeyT', bound=Hashable)
 
+# ##### Regex for Decimal JSON Schema Generation #####
+_DECIMAL_JSON_VALIDATION_PATTERN = r"""
+    ^
+    -?                            # Minus sign (optional)
+    \d+                           # One or more digits for the integer part (allows leading zeros)
+    (                             # Optional decimal group
```

Comment:
Can we make it non capturing? (same for the other regex). Or is there a reason not to do so?